### PR TITLE
[examples]educe memory consumption

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

accessing the whole DistTensor is quite memory consuming. Let's remove unnecessary ones.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/graphsage/dist/train_dist.py [hunk 1]
```diff
@@ -307,7 +309,7 @@ def main(args):
     parser.add_argument('--ip_config', type=str, help='The file for IP configuration')
     parser.add_argument('--part_config', type=str, help='The path to the partition config file')
     parser.add_argument('--num_clients', type=int, help='The number of clients')
-    parser.add_argument('--n_classes', type=int, help='the number of classes')
+    parser.add_argument('--n_classes', type=int, default=0, help='the number of classes')
```
Comment (jermainewang) [inline examples/pytorch/graphsage/dist/train_dist.py:312]: may be -1 is a better default value.

Please tell what the default value means and what is the impact in the help message.

**Review by jermainewang — APPROVED** [review]

File: examples/pytorch/graphsage/dist/train_dist.py [hunk 2]
```diff
@@ -307,8 +309,11 @@ def main(args):
     parser.add_argument('--ip_config', type=str, help='The file for IP configuration')
     parser.add_argument('--part_config', type=str, help='The path to the partition config file')
     parser.add_argument('--num_clients', type=int, help='The number of clients')
-    parser.add_argument('--n_classes', type=int, help='the number of classes')
-    parser.add_argument('--backend', type=str, default='gloo', help='pytorch distributed backend')
+    parser.add_argument('--n_classes', type=int, default=-1,
+                        help='the number of classes. If not specified, this value will be calculated'
```
Comment (frozenbugs) [inline examples/pytorch/graphsage/dist/train_dist.py:313]: Capitalize the T.
and let's aim for 80 for the max line length.

Comment (Rhett-Ying) [reply examples/pytorch/graphsage/dist/train_dist.py:313]: `80` seems a bit small. in this case, many blanks occupies in each line, so we will have many lines for this help message?

Comment (frozenbugs) [reply examples/pytorch/graphsage/dist/train_dist.py:313]: Yes, it is fine to have multiple lines if the code is nicely laid out.

**Pull request merged**